pinot-segment-local/src/main/java/org/apache/pinot/segment/local/indexsegment/immutable/ImmutableSegmentImpl.java

```diff
@@ -159,7 +159,11 @@ public void deleteValidDocIdsSnapshot() {
     }
   }
 
-  private File getValidDocIdsSnapshotFile() {
+  private File getValidDocIdsSnapshotFile(boolean isTTLEnabled) {
+    if (isTTLEnabled) {
+      return new File(SegmentDirectoryPaths.findSegmentDirectory(_segmentMetadata.getIndexDir()),
```

Comment:
what is the need for a seperate file? Both are doing the same thing right from the load perspective?

Reply:
thanks @KKcorps The reason to have a separate file is that We should mark snapshot persisted by TTL differently from the regular snapshot, so validDocsIdsSnapshot from upsertTTL will not be override by SnapshotEnabled feature. We can also use snapshot for segment compaction and other upsert improvements.

Another potential solution is disable `enableSnapshot` for upsert TTL so that both snapshots won't overwrite each other.

Reply:
Makes sense. Then I want to understand, how does this process work during server restart because we will be loading validDocIds from the snapshot file and then discarding them based on this ttl_snapshot file?

Also, what happens if all docs of the segment have been removed due to TTL and we restart the server?

Reply:
@KKcorps thanks for the code review and pointing out this part. 
I think I missed this part in the code, `loadValidDocIdsFromSnapshot(true)` and `deleteValidDocIdsSnapshot(true)` should be called here when TTLConfig exists. I will add it.

I also think about this part again, I feel snapshot feature and TTLConfig  and enableSnapshot might not need to be enabled together, otherwise we will store duplicated snapshots, and the snapshot persisted by the snapshotEnabled will never be used by TTL feature. If we go through this route, we will handle three scenarios 
1) TTL config enabled
2) snapshot enabled
3) both are not enabled.

I will also add validation to avoid both config enabled in upsertConfig.

Reply:
With this change, the load and deletion behavior will be the same for snapshotEnabled and upsertTTL upon server restart, however how to persist snaphost is different for the above two features.